src/abstractdataframe/show.jl

```diff
@@ -543,32 +549,68 @@ end
 #' df = DataFrame(A = 1:3, B = ["x", "y", "z"])
 #' showall(df, true)
 function Base.showall(df::AbstractDataFrame,
-                      splitchunks::Bool = false) # -> Void
-    showall(STDOUT, df, splitchunks)
+                      allcols::Bool = true) # -> Void
+    showall(STDOUT, df, allcols)
     return
 end
 
+#' @exported
 #' @description
 #'
 #' Render a summary of the column names, column types and column missingness
 #' count.
 #'
 #' @param io::IO The `io` to be rendered to.
 #' @param df::AbstractDataFrame An AbstractDataFrame.
+#' @param allcols::Bool If `false` (default), only a subset of columns
+#'        fitting on the screen is printed.
+#' @param values::Bool If `true` (default), first and last value of
+#'        each column is printed.
 #'
 #' @returns o::Void A `nothing` value.
 #'
 #' @examples
 #'
 #' df = DataFrame(A = 1:3, B = ["x", "y", "z"])
-#' showcols(df, true)
-function showcols(io::IO, df::AbstractDataFrame) # -> Void
+#' showcols(STDOUT, df)
+function showcols(io::IO, df::AbstractDataFrame, allcols::Bool = false, values::Bool = true) # -> Void
     println(io, summary(df))
     metadata = DataFrame(Name = _names(df),
                          Eltype = eltypes(df),
                          Missing = colmissing(df))
-    showall(io, metadata, true, Symbol("Col #"), false)
+    nrows, ncols = size(df)
+    if values && nrows > 0
+        if nrows == 1
+            metadata[:Values] = [Symbol(sprint(showcompact, df[1, i])) for i in 1:ncols]
+        else
+            metadata[:Values] = [Symbol(sprint(showcompact, df[1, i]),
+                                        "  …  ",
+                                        sprint(showcompact, df[end, i])) for i in 1:ncols]
+        end
+    end
+    (allcols?showall:show)(io, metadata, true, Symbol("Col #"), false)
     return
 end
 
-showcols(df::AbstractDataFrame) = showcols(STDOUT, df) # -> Void
+#' @exported
+#' @description
+#'
+#' Render a summary of the column names, column types and column missingness
+#' count.
+#'
+#' @param df::AbstractDataFrame An AbstractDataFrame.
+#' @param allcols::Bool If `false` (default), only a subset of columns
```

Comment:
Maybe just call this `all`, since "col" is already clear from the function's name?

Reply:
This still applies, right?